### ISSUE TITLE
model.refresh is not working properly | bug

### ISSUE THREAD

**Issue by seeden** [seegno/bookshelf-json-columns#42]
When I call model.refresh() I will get next exception:

```
SyntaxError: Unexpected token o in JSON at position 1
    at Object.parse (native)
    at constructor.jsonColumns.forEach.column (/Users/seeden/Documents/git/goodmodule.com/node_modules/bookshelf-json-columns/dist/index.js:38:38)
    at Array.forEach (native)
    at Quiz.parse (/Users/seeden/Documents/git/goodmodule.com/node_modules/bookshelf-json-columns/dist/index.js:36:32)
    at /Users/seeden/Documents/git/goodmodule.com/node_modules/bookshelf/lib/base/events.js:176:25
From previous event:
    at Quiz.triggerThen (/Users/seeden/Documents/git/goodmodule.com/node_modules/bookshelf/lib/base/events.js:175:32)
    at Quiz.<anonymous> (/Users/seeden/Documents/git/goodmodule.com/node_modules/bookshelf/lib/model.js:702:19)
    at tryOnImmediate (timers.js:543:15)
    at processImmediate [as _immediateCallback] (timers.js:523:5)
From previous event:
    at Quiz.<anonymous> (/Users/seeden/Documents/git/goodmodule.com/node_modules/bookshelf/lib/model.js:685:8)
From previous event:
    at Quiz.refresh (/Users/seeden/Documents/git/goodmodule.com/node_modules/bookshelf/lib/model.js:567:17)
```

Instead of this everything else working fine.

bookshelf-json-columns@2.0.0 with sqllite

**Comment by ricardogama** [comment]
Hi @seeden.

It seems the value is already parsed when the plugin `fetched` event is fired, so I suspect that is being done somewhere else. Do you have any custom events on you model or are you extending `initialize` or `save` methods?

It can also happens that the value comes already parsed from the fetch, which is unlikely, but I wonder if the model data you are refreshing was inserted with the plugin or with a raw query for example.

**Comment by seeden** [comment]
I am not extending initialize or save method

**Comment by ricardogama** [comment]
I can't reproduce this error, the only thing that occurs to me is that the fetched value is not a valid stringified JSON object. Can debug what value is being parsed when the error occurs? 

Also, a snippet of your model and the code you're executing would help, along with the **bookshelf** and **sqlite3** versions you're using.

**Comment by seeden** [comment]
This issue is happing when I use increment. Here is simple example

```js
const model = Model.where({id}).fetch();
model.query().increment('likes', 1);
model.refresh();
```

Your package is using parsed data.

**Comment by seeden** [comment]
I am using temp fix for this 
```js
this.attributes[column] = typeof this.attributes[column] === 'string'
   ? JSON.parse(this.attributes[column])
   : this.attributes[column];
```

**Comment by ricardogama** [comment]
Ok I got the bug, it happens when querying updates or inserts through knex, which does not trigger the `saving` event and only triggers `fetching`. 

I'll try to fix until the end of the week, thanks for the info!

**Comment by ricardogama** [comment]
Released as [2.0.1](https://github.com/seegno/bookshelf-json-columns/releases/tag/2.0.1).
